beacon-chain/core/epoch/precompute/slashing_test.go

```diff
@@ -0,0 +1,123 @@
+package precompute
+
+import (
+	"testing"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	ethpb "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+func TestProcessSlashingsPrecompute_NotSlashed(t *testing.T) {
+	s := &pb.BeaconState{
+		Slot:       0,
+		Validators: []*ethpb.Validator{{Slashed: true}},
+		Balances:   []uint64{params.BeaconConfig().MaxEffectiveBalance},
+		Slashings:  []uint64{0, 1e9},
+	}
+	bp := &Balance{CurrentEpoch: params.BeaconConfig().MaxEffectiveBalance}
+	newState := ProcessSlashingsPrecompute(s, bp)
+	wanted := params.BeaconConfig().MaxEffectiveBalance
+	if newState.Balances[0] != wanted {
+		t.Errorf("Wanted slashed balance: %d, got: %d", wanted, newState.Balances[0])
+	}
+}
+
+func TestProcessSlashingsPrecompute_SlashedLess(t *testing.T) {
+
+	tests := []struct {
+		state *pb.BeaconState
+		want  uint64
+	}{
+		{
+			state: &pb.BeaconState{
+				Validators: []*ethpb.Validator{
+					{Slashed: true,
+						WithdrawableEpoch: params.BeaconConfig().EpochsPerSlashingsVector / 2,
+						EffectiveBalance:  params.BeaconConfig().MaxEffectiveBalance},
+					{ExitEpoch: params.BeaconConfig().FarFutureEpoch, EffectiveBalance: params.BeaconConfig().MaxEffectiveBalance}},
+				Balances:  []uint64{params.BeaconConfig().MaxEffectiveBalance, params.BeaconConfig().MaxEffectiveBalance},
+				Slashings: []uint64{0, 1e9},
+			},
+			// penalty    = validator balance / increment * (3*total_penalties) / total_balance * increment
+			// 3000000000 = (32 * 1e9)        / (1 * 1e9) * (3*1e9)             / (32*1e9)      * (1 * 1e9)
+			want: uint64(29000000000), // 32 * 1e9 - 3000000000
```

Comment:
Please update the comments. This math doesn't check out

Reply:
edit nvm it does (?) i think

Reply:
Yea the math is correct